app/src/main/java/org/p2p/wallet/jupiter/ui/main/JupiterSwapPresenter.kt

```diff
@@ -181,6 +188,24 @@ class JupiterSwapPresenter(
                     stateManager.onNewAction(SwapStateAction.CancelSwapLoading)
                     val transactionState = TransactionState.JupiterSwapSuccess
                     transactionManager.emitTransactionState(internalTransactionId, transactionState)
+
+                    val pendingTransaction = RpcHistoryTransaction.Swap(
```

Comment:
Maybe extract this big constructor to companion object and pass only currentState and result?